akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -39,10 +39,26 @@ object ActorTestKit {
    */
   def apply(): ActorTestKit =
     new ActorTestKit(
-      name = TestKitUtils.testNameFromCallStack(classOf[ActorTestKit]),
-      config = ApplicationTestConfig,
+      ActorSystem(
+        ActorTestKitGuardian.testKitGuardian,
+        TestKitUtils.testNameFromCallStack(classOf[ActorTestKit]),
+        ApplicationTestConfig),
       settings = None)
 
+  /**
+   * Create a testkit from the provided actor system.
+   *
+   * It will use the provided [[akka.actor.typed.ActorSystem]] and create a ActorTestKit,
+   * e.g. threads will include the name of the provided actor system.
+   * When the test has completed you should terminate the `ActorSystem` and
+   * the testkit with [[ActorTestKit#shutdownTestKit]].
+   *
+   * Config loaded from `system.settings.config` if that exists, otherwise
+   * using default configuration from the reference.conf resources that ship with the Akka libraries.
+   */
+  def apply(system: ActorSystem[_]): ActorTestKit =
```

Comment:
we could also add corresponding to `ScalaTestWithActorTestKit` and `TestKitJunitResource` so an ActorSystem can be passed in also when using those